numpy/core/include/numpy/npy_3kcompat.h

```diff
@@ -167,6 +167,24 @@ static inline int PyInt_Check(PyObject *op) {
 
 #endif /* NPY_PY3K */
 
+/*
+ * Macros to protect CRT calls against instant termination when passed an
+ * invalid parameter (issue23524).
+ */
+#if defined _MSC_VER && _MSC_VER >= 1900
+
+extern _invalid_parameter_handler _Py_silent_invalid_parameter_handler;
+#define NPY_BEGIN_SUPPRESS_IPH { _invalid_parameter_handler _Py_old_handler = \
+    _set_thread_local_invalid_parameter_handler(_Py_silent_invalid_parameter_handler);
```

Comment:
This looked a little fishy to me at first, but this function is documented by MSVC even though it's apparently private: https://learn.microsoft.com/en-us/cpp/c-runtime-library/reference/set-invalid-parameter-handler-set-thread-local-invalid-parameter-handler?view=msvc-170.

One question: it's supposed to come from `stdlib.h`, so isn't it better to explicitly include that here? Like in CPython: https://github.com/python/cpython/blob/36854bbb240e417c0df6f0014924fcc899388186/Include/internal/pycore_fileutils.h#L264

Reply:
Line 15 has `#include <Python.h>` which has conditionally [`#include <stdlib.h>`](https://github.com/python/cpython/blob/36854bbb240e417c0df6f0014924fcc899388186/Include/Python.h#L23), which is why this works for me. It would be better to explicitly include it. I will add that.